flink-core/src/main/java/org/apache/flink/api/common/typeinfo/Types.java

```diff
@@ -19,56 +19,408 @@
+	 *
+	 * <p>If Flink's type analyzer is unable to extract a valid POJO type information with
+	 * type information for all fields, an {@link org.apache.flink.api.common.functions.InvalidTypesException}
+	 * is thrown.
+	 *
+	 * @param pojoClass POJO class to be analyzed by Flink
+	 */
+	public static TypeInformation<?> POJO(Class<?> pojoClass) {
+		final TypeInformation<?> ti = TypeExtractor.createTypeInfo(pojoClass);
+		if (ti instanceof PojoTypeInfo) {
+			return ti;
+		}
+		throw new InvalidTypesException("POJO type expected but was: " + ti);
+	}
+
+	/**
+	 * Returns type information for a POJO (Plain Old Java Object) and allows to specify all fields manually.
+	 *
+	 * <p>A POJO class is public and standalone (no non-static inner class). It has a public no-argument
+	 * constructor. All non-static, non-transient fields in the class (and all superclasses) are either public
+	 * (and non-final) or have a public getter and a setter method that follows the Java beans naming
+	 * conventions for getters and setters.
+	 *
+	 * <p>A POJO is a fixed-length, null-aware composite type with non-deterministic field order. Every field
+	 * can be null independent of the field's type.
+	 *
+	 * <p>Types for all fields of the POJO can be defined in a hierarchy of subclasses.
+	 *
+	 * <p>If Flink's type analyzer is unable to extract a POJO field, an
+	 * {@link org.apache.flink.api.common.functions.InvalidTypesException} is thrown.
+	 *
+	 * <p><strong>Note:</strong> In most cases the type information of fields can be determined automatically,
+	 * we recommend to use {@link Types#POJO(Class)}.
+	 *
+	 * @param pojoClass POJO class
+	 * @param fields map of fields that map a name to type information
+	 */
+	public static TypeInformation<?> POJO(Class<?> pojoClass, Map<String, TypeInformation<?>> fields) {
+		final List<PojoField> pojoFields = new ArrayList<>(fields.size());
+		for (Map.Entry<String, TypeInformation<?>> field : fields.entrySet()) {
+			Field f = TypeExtractor.getDeclaredField(pojoClass, field.getKey());
+			if (f == null) {
+				throw new InvalidTypesException("Field '" + field.getKey() + "'could not be accessed.");
+			}
+			pojoFields.add(new PojoField(f, field.getValue()));
+		}
+
+		return new PojoTypeInfo<>(pojoClass, pojoFields);
+	}
+
+	/**
+	 * Returns generic type information for any Java object. The serialization logic will
+	 * use the general purpose serializer Kryo.
+	 *
+	 * <p>Generic types are black-boxes for Flink, but allow any object and null values in fields.
+	 *
+	 * <p>By default, serialization of this type is not very efficient. Please read the documentation
+	 * about how to improve efficiency (namely by pre-registering classes).
+	 *
+	 * @param genericClass any Java class
+	 */
+	public static TypeInformation<?> GENERIC(Class<?> genericClass) {
+		return new GenericTypeInfo<>(genericClass);
+	}
+
+	/**
+	 * Returns type information for Java arrays of primitive type (such as <code>byte[]</code>). The array
```

Comment:
`The array and its elements do not support null values.` -> `The array must not be null.`? Entries of primitive arrays cannot be null anyway.

Reply:
I just wanted to be consistent :D